android/build.gradle

```diff
@@ -0,0 +1,224 @@
+        implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
+        testImplementation 'junit:junit:4.12'
+        implementation "net.java.dev.jna:jna:$jna_version@aar"
+    }
+}
+
+ext {
+    rustTargets = [
+//        'arm',
+//        'arm64',
+//        'x86_64', Commented out for testability
+'x86',
+    ]
+
+}
+
+cargo {
+    // The directory of the Cargo.toml to build.
+    module = '../experiments'
+
+    // The Android NDK API level to target.
+    apiLevel = 29
+
+    // Where Cargo writes its outputs.
+    targetDirectory = '../target'
+
+    libname = 'nimbus'
+
+    // The Cargo targets to invoke.  The mapping from short name to target
+    // triple is defined by the `rust-android-gradle` plugin.
+    targets = rustTargets
+
+    // Perform release builds (which should have debug info, due to
+    // `debug = true` in Cargo.toml).
+    profile = "release"
+
+    extraCargoBuildArguments = []
+}
+
+
+afterEvaluate {
+    // The `cargoBuild` task isn't available until after evaluation.
+    android.libraryVariants.all { variant ->
+        def productFlavor = ""
+        variant.productFlavors.each {
+            productFlavor += "${it.name.capitalize()}"
+        }
+        def buildType = "${variant.buildType.name.capitalize()}"
+        tasks["merge${productFlavor}${buildType}JniLibFolders"].dependsOn(tasks["cargoBuild"])
+    }
+}
+
+// We also do the binding generation here:
+android.libraryVariants.all { variant ->
+    def t = tasks.register("generate${variant.name.capitalize()}UniffiBindings", Exec) {
+        workingDir "${project.projectDir}"
+        commandLine 'uniffi-bindgen', 'generate', '../experiments/src/nimbus.idl', '--language', 'kotlin', '--out-dir', "${buildDir}/generated/source/uniffi/${variant.name}/java"
+    }
+    variant.javaCompileProvider.get().dependsOn(t)
+    def sourceSet = variant.sourceSets.find { it.name == variant.name }
+    sourceSet.java.srcDir new File(buildDir, "generated/source/uniffi/${variant.name}/java")
+    // XXX: I've been trying to make this work but I can't, so the compiled bindings will show as "regular sources" in Android Studio.
+    idea.module.generatedSourceDirs += file("${buildDir}/generated/source/uniffi/${variant.name}/java/uniffi")
+}
+
+allprojects {
+    group = "org.mozilla.nimbus"
+    apply plugin: 'digital.wup.android-maven-publish'
+
+    // This allows to invoke Gradle like `./gradlew publishToRootProjectBuildDir` (equivalent to
+    // `./gradlew publish`) and also `./gradlew publishToProjectBuildDir`.
+    publishing {
+        repositories {
+            maven {
+                name = "rootProjectBuildDir"
+                url "file://${project.rootProject.buildDir}/maven"
+            }
+            maven {
+                name = "projectBuildDir"
+                url "file://${project.buildDir}/maven"
+            }
+        }
+    }
+}
+
+ext.configurePublish = { jnaForTestConfiguration = null ->
```

Comment:
This should probably go into a `publish.gradle` like we do in other projects

Reply:
I'm not too familiar with the details of the publishing,

But we'd need to add a .buildconfig to retrieve the values that I have hard coded here

(i.e the name of the package, the version etc)